### PR TITLE
start add unit test

### PR DESCRIPTION
**Pull request by antarus**
correct lsint error. 
I don't understand the last error :
 error  Don't make functions within a loop  no-loop-func

I need a reference to generator.log and generator.fs how can i obtain a reference to the main generator from test?
disable pretest script for my test...

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
@antarus : sorry, I dont have the answer to your 2 questions.

I'm not a NodeJS developer. Maybe 2 links which can help you to understand (you probably already searched them):
- https://eslint.org/docs/rules/no-loop-func
- https://stackoverflow.com/questions/13082302/js-hint-dont-make-functions-within-a-loop

About generator.log, same, I have no idea

**Comment by pascalgrimaud** [comment]
I will fix the Travis tests for gulp eslint, directly to master, as it should failed here

**Comment by antarus** [comment]
:( who is nodejs developer?
Thanks for the links but I already had to watch and I do not understand.
I will look at the different modules to see if I find a solution to have a ref of the main generator..

**Comment by pascalgrimaud** [comment]
Maybe @deepu105 or @MathieuAA you know the answers, to help this new contributor ?

**Comment by antarus** [comment]
Thanks I found to access the main generator by looking at other module;)
By cons the error lint ...

File: generators/app/utils.js [hunk 1]
```diff
@@ -135,12 +133,13 @@ function diffArray(arr1, arr2, generator) {
             if (arr1 === undefined || arr2 === undefined) {
                 return undefined;
             }
-            reduc = _.reduce(arr1, function (result, value, key) {
-                if (arr2[key]) {
-                    return _.isEqual(value, arr2[key]) ? result : result.concat(key);
-                }
-                return false;
-            }, []);
+            // reduc = _.reduce(arr1, (result, value, key) => {
```
Comment (MathieuAA) [inline generators/app/utils.js]: @antarus @pascalgrimaud Hi guys, just so you know why the linter yelled: there was a declared function here, hence the warning. I admit it's a bit tricky to understand but having this kind of code results in having two (or more, depends on the implem) nested loops:
  - the while
  - the reduce
  - maybe the isEqual (I don't yet know the chosen implem of that one)

**Comment by MathieuAA** [comment]
@pascalgrimaud sorry, didn't have my notifications on yesterday =/

**Comment by pascalgrimaud** [comment]
No problem @MatthieuAA it seems resolved !
Thanks :-)

**Pull request merged**